src/sonic-py-common/sonic_py_common/security_cipher.py

```diff
@@ -0,0 +1,120 @@
+'''
+
+A common module for handling the encryption and
+decryption of the feature passkey. It also takes
+care of storying the secure cipher at root 
+protected file system
+
+'''
+
+import subprocess
+import threading
+import syslog
+import os
+from swsscommon.swsscommon import ConfigDBConnector
+
+class security_cipher:
+    _instance = None
+    _lock = threading.Lock()
+
+    def __new__(cls):
+        with cls._lock:
+            if cls._instance is None:
+                cls._instance = super(security_cipher, cls).__new__(cls)
+                cls._instance._initialized = False
+        return cls._instance
+
+    def __init__(self):
+        if not self._initialized:
+            self._file_path = "/etc/cipher_pass"
+            self._config_db = ConfigDBConnector()
+            self._config_db.connect()
+            # Note: Kept 1st index NA intentionally to map it with the cipher_pass file
+            # contents. The file has a comment at the 1st row / line
+            self._feature_list = ["NA", "TACPLUS", "RADIUS", "LDAP"]
+            if not os.path.exists(self._file_path):
+                with open(self._file_path, 'w') as file:
+                    file.writelines("#Auto generated file for storing the encryption passwords\n")
+                    file.writelines("TACPLUS : \nRADIUS : \nLDAP :\n")
+                    #os.chmod(self._file_path, 0o644)
+            self._initialized = True
+
+    # Write cipher_pass file
+    def __write_passwd_file(self, feature_type, passwd):
+        if feature_type == 'NA':
+            syslog.syslog(syslog.LOG_ERR, "__write_passwd_file: Invalid feature type: {}".format(feature_type))
+            return
+
+        if feature_type in self._feature_list:
+            try:
+                with open(self._file_path, 'r') as file:
+                    lines = file.readlines()
+                    # Update the password for given feature
+                    lines[self._feature_list.index(feature_type)] = feature_type + ' : ' + passwd + '\n'
+
+                    #os.chmod(self._file_path, 0o777)
+                    with open(self._file_path, 'w') as file:
+                        file.writelines(lines)
```

Comment:
Is this 'root' only accessible file?

Reply:
Yes it is.

`root@sonic:/tmp# ls /etc/cipher_pass -l

-rw-r--r-- 1 root root 102 Nov 16 16:44 /etc/cipher_pass`

Reply:
Why is it globally readable?
Shouldn't we restrict this to:
-rw-r-----
At the very least?

Reply:
Yeah, it is doable but will it make much difference? config_db.json is too globally readable, isn't it? :-)

Reply:
@Yarden-Z As requested, updated the access permission from 644 to 640.
```
sonic:~# ls /etc/cipher_pass -l
-rw-r----- 1 root root 91 Feb  1 01:18 /etc/cipher_pass 
```